examples/v3/multipart/consumer.spec.ts

```diff
@@ -0,0 +1,231 @@
+        uponReceiving: 'a multipart file upload request',
+        withRequest: {
+          method: 'POST',
+          path: '/upload',
+          body: form, // The multipart form data as a string (including file content)
+          headers, // Headers including Content-Type with boundary
+        },
+        willRespondWith: {
+          status: 200,
+          body: {
+            // Use matchers to flexibly match any values in the response
+            success: MatchersV3.boolean(true),
+            message: MatchersV3.like('File uploaded successfully'),
+          },
+        },
+      })
+      // Execute the test against the mock server
+      .executeTest(async (mockServer) => {
+        // Send the actual multipart file upload request
+        const response = await axios.post(`${mockServer.url}/upload`, form, {
+          headers,
+        });
+
+        // Verify the response indicates success
+        expect(response.data.success).to.equal(true);
+        expect(response.data.message).to.equal('File uploaded successfully');
+
+        // Clean up the temporary file
+        fs.unlinkSync(tmpFile);
+      });
+  });
+
+  it('creates a pact using withRequestMultipartFileUpload helper method', async () => {
+    // Use the pre-existing test file in the multipart directory
+    const path = require('path');
+    const testFile = path.join(__dirname, 'test-upload-file.txt');
+
+    await pact
+      .given('a file upload is expected')
+      .uponReceiving(
+        'a multipart file upload using withRequestMultipartFileUpload'
+      )
+      .withRequestMultipartFileUpload(
+        {
+          method: 'POST',
+          path: '/upload-file',
+        },
+        'text/plain', // The content type of the file being uploaded
+        testFile,
+        'document' // The name of the form field for the file
+      )
+      .willRespondWith({
+        status: 200,
+        body: {
+          success: MatchersV3.boolean(true),
+          filename: MatchersV3.like('test-upload-file.txt'),
+          message: MatchersV3.like('File received successfully'),
+        },
+      })
+      .executeTest(async (mockServer) => {
+        // Create form data with the file
+        const fs = require('fs');
+        const formData = new FormData();
+        const fileContent = fs.readFileSync(testFile);
+        formData.append('document', fileContent, 'test-upload-file.txt');
+
+        // Send the request
+        const response = await axios.post(
+          `${mockServer.url}/upload-file`,
+          formData,
+          {
+            headers: formData.getHeaders(),
+          }
+        );
+
+        // Verify the response
+        expect(response.data.success).to.equal(true);
+        expect(response.data.filename).to.equal('test-upload-file.txt');
+        expect(response.data.message).to.equal('File received successfully');
+      });
+  });
+
+  it('creates a pact using withRequestMultipartFileUpload with custom boundary', async () => {
```

Comment:
And this example should explain in what circumstances we would want a custom boundary, pros and cons, etc.